app/controllers/BankDetailsController.scala

```diff
@@ -98,7 +98,19 @@ class BankDetailsController @Inject() (
   }
 
   val checkBankDetails: Action[AnyContent] = as.eligibleJourneyAction { implicit request =>
-    Ok("This is where the check bank details page will go, for now, here's the journey:\n\n\n" + Json.prettyPrint(Json.toJson(request.journey)))
+    request.journey match {
+      case j: Journey.BeforeEnteredDirectDebitDetails => JourneyIncorrectStateRouter.logErrorAndRouteToDefaultPage(j)
+      case j: Journey.AfterEnteredDirectDebitDetails  => displayCheckBankDetailsPage(j)
```

Comment:
we could put an extra guard here on the stage to see if the user said that they were the sole signatory - if they said no then we shouldn't show this page

Reply:
Sure, will update

Reply:
Done, I added a test for this too